Airwallex/ApplePay/AWXApplePayProvider.m

```diff
@@ -109,8 +138,17 @@ - (void)paymentAuthorizationControllerDidFinish:(nonnull PKPaymentAuthorizationC
                 [strongSelf completeWithResponse:strongSelf.lastResponse error:strongSelf.lastError];
             });
         } else {
-            // The user most likely has cancelled the authorization.
-            // Do nothing here to allow the user to select another payment method.
+            if (strongSelf.isApplePayInitiatedDirectly) {
+                dispatch_async(dispatch_get_main_queue(), ^{
+                    NSError *error = [NSError errorWithDomain:AWXSDKErrorDomain
+                                                         code:-1
+                                                     userInfo:@{NSLocalizedDescriptionKey: NSLocalizedString(@"User cancelled Apple Pay.", nil)}];
+                    [[self delegate] provider:self didCompleteWithStatus:AirwallexPaymentStatusCancel error:error];
```

Comment:
I think we want to put `AirwallexPaymentStatusFailure` status here since error message should be specifically for that, although it's loosely coupled here. Also just to align with other `User cancelled` scenario in `AWX3DSViewController`, which also returns Failure status in the end.

Reply:
Isn't that the wrong status message to send failure when user has manually cancelled the payment? 🤔

Reply:
Removed error message and just returning cancel status as per our discussion